### PR TITLE
Adding lepton SF (from the ttH analysis)

### PR DESCRIPTION
**Pull request by GonzalezFJR**
Using SF from the ttH analysis as a placeholder. They should be close to the final ones and should help to obtain a better data/MC agreement.

### CODE REVIEW

File: analysis/topEFT/topeft.py [hunk 1]
```diff
@@ -277,8 +289,21 @@ def process(self, events):
...
+          weights[r].add('norm',genw if isData else (xsec/sow)*genw)
+
+        weights['ee'].add('lepSF_eeSS', lepSF_eeSS)
+        weights['em'].add('lepSF_emSS', lepSF_emSS)
+        weights['mm'].add('lepSF_mmSS', lepSF_mumuSS)
+        weights['eee'].add('lepSF_eee', lepSF_eee)
+        weights['mmm'].add('lepSF_mmm', lepSF_mmm)
+        weights['mme'].add('lepSF_mme', lepSF_mme)
+        weights['eem'].add('lepSF_eem', lepSF_eem)
+
```
Comment (bryates) [inline analysis/topEFT/topeft.py:306]: Maybe we should fold this into the for loop above, and just add a lepSF if r isn't all?

**Comment by bryates** [comment]
Is https://github.com/TopEFT/topcoffea/blob/9ee2f35e21e04535265533406e691fb9f67f5df5/topcoffea/cfg/2018.cfg updating to the latest samples, or just the ones that were used to test this branch?

File: topcoffea/modules/objects.py [hunk 2]
```diff
@@ -50,7 +50,7 @@ def isCleanJet(jets, electrons, muons, taus, drmin=0.4):
   return (jetEleMask & jetMuMask & jetTauMask)
 
 def isTightJet(pt, eta, jet_id, neHEF, neEmEF, chHEF, chEmEF, nConstituents, jetPtCut=30.0):
-    mask = (pt>jetPtCut) & (abs(eta)<2.4)# & ((jet_id&6)==6)
+    mask = (pt>jetPtCut) & (abs(eta)<2.4) & ((jet_id&6)==6)
```
Comment (bryates) [inline topcoffea/modules/objects.py:53]: Will this break with data, or is `jet_id` set by the jet tagging algorithm?

File: topcoffea/cfg/2018.cfg [hunk 3]
```diff
@@ -7,82 +7,82 @@ verbose: 1
...
+#
+##MuonEG
+#MuonEG_2018 : MuonEG_Run2018A_Nano25Oct2019_v1, MuonEG_Run2018B_Nano25Oct2019_v1, MuonEG_Run2018C_Nano25Oct2019_v1, MuonEG_Run2018D_Nano25Oct2019_ver2_v1 : 10
+#
+##DoubleMuon
+#DoubleMuon_2018 : DoubleMuon_Run2018A_Nano25Oct2019_v1, DoubleMuon_Run2018B_Nano25Oct2019_v1, DoubleMuon_Run2018C_Nano25Oct2019_v1, DoubleMuon_Run2018D_Nano25Oct2019_ver2_v1 : 10
+#
+## Single Muon
+#SingleMuon_2018 : SingleMuon_Run2018A_Nano25Oct2019_v1, SingleMuon_Run2018B_Nano25Oct2019_v1, SingleMuon_Run2018C_Nano25Oct2019_v1, SingleMuon_Run2018D_Nano25Oct2019_v1 : 10
+#
```
Comment (bryates) [inline topcoffea/cfg/2018.cfg:88]: Is updating to the latest samples, or just the ones that were used to test this branch?

File: analysis/topEFT/topeft.py (hunk 1 above)
Comment (GonzalezFJR) [reply analysis/topEFT/topeft.py:306]: Well, then we'd need to build a dict or a list with the lepSF arrays so they can be pointed to in the loop.

File: topcoffea/modules/objects.py (hunk 2 above)
Comment (GonzalezFJR) [reply topcoffea/modules/objects.py:53]: jet_id is a reco variable, present in data

File: topcoffea/cfg/2018.cfg (hunk 3 above)
Comment (GonzalezFJR) [reply topcoffea/cfg/2018.cfg:88]: hmm right, 2018.cfg is pointing to local files so shouldn't probably be in the commit. I'll remove it.

File: analysis/topEFT/topeft.py (hunk 1 above)
Comment (bryates) [reply analysis/topEFT/topeft.py:306]: Good point, we'll leave it as is.

**Pull request merged**